src/hooks/use-aggrid-row-filter-impl.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { AgGridReact } from 'ag-grid-react';
+import React, { useCallback, useEffect } from 'react';
+import {
+    FilterSelectorType,
+    UseAggridRowFilterOutputType,
+    useAggridRowFilter,
+} from './use-aggrid-row-filter';
+import { FILTER_DATA_TYPES } from 'components/custom-aggrid/custom-aggrid-header.type';
+
+export const useAggridRowFilterImpl = (
+    gridRef: React.MutableRefObject<AgGridReact | null>,
+    filterSelectorKeys: Record<string, string>,
+    updateFilterCallback = () => {}
+): UseAggridRowFilterOutputType => {
+    const { updateFilter, filterSelector, initFilters } = useAggridRowFilter(
+        filterSelectorKeys,
+        updateFilterCallback
+    );
+
+    const formatCustomFiltersForAgGrid = useCallback(
+        (filters: FilterSelectorType[]) => {
+            const agGrifFilterModel: Record<string, any> = {};
```

Comment:
```suggestion
            const agGridFilterModel: Record<string, any> = {};
```

Reply:
I think we should change "Record<string, any>" to either : 
- "any" -> it's the type expected "setFilterModel"
- "Record<string, FilterType>" with FilterType a new TypeScript interface with "filterType", "type" and "filter" properties

Reply:
`setFilterModel = (
    model: FilterModel | null
) => void;

interface FilterModel {
  [colId: string]: any;
}`
this is from the documentation of AgGrid https://www.ag-grid.com/react-data-grid/filter-api/#get--set-all-filter-models

Reply:
I see, when you check "gridApi.d.ts", it's not the case 
`setFilterModel(model: any): void;`

If it's documented like this, let's keep it this way